crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,41 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&mut self, conditional: TransactionInterop);
+
+    /// Get attached [`TransactionInterop`] if any.
+    fn interop(&self) -> Option<&TransactionInterop>;
+
+    /// Helper that sets the conditional and returns the instance again
+    fn with_interop(mut self, interop: TransactionInterop) -> Self
+    where
+        Self: Sized,
+    {
+        self.set_interop(interop);
+        self
+    }
+}
+
+/// Helper to keep track of cross transaction interop validity
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct TransactionInterop {
```

Comment:
not sure if op-geth already has terminology for this but we could name this TransactionInteropContext, TransactionInteropMeta

Reply:
We have TransactionConditional, so i kept the style